### PR TITLE
[Enhancement] Block creating db using storage volume in shared-nothing mode

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateDbAnalyzer.java|4|5|80.00%|[51]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateDbAnalyzer.java [hunk 1]
```diff
@@ -38,5 +43,14 @@ public static void analyze(CreateDbStmt statement, ConnectContext context) {
         } catch (AnalysisException e) {
             ErrorReport.reportSemanticException(ErrorCode.ERR_BAD_CATALOG_ERROR, catalogName);
         }
+
+        Map<String, String> properties = statement.getProperties();
+        if (properties.containsKey(PropertyAnalyzer.PROPERTIES_STORAGE_VOLUME)) {
+            String volume = properties.get(PropertyAnalyzer.PROPERTIES_STORAGE_VOLUME);
+            if (RunMode.getCurrentRunMode() == RunMode.SHARED_NOTHING && !volume.equals(StorageVolumeMgr.LOCAL)) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateDbAnalyzer.java:50]:
```suggestion
            if (RunMode.getCurrentRunMode() == RunMode.SHARED_NOTHING && !StorageVolumeMgr.LOCAL.equalsIgnoreCase(volume) {
```

**Review by wyb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateDbAnalyzer.java [hunk 2]
```diff
@@ -38,5 +43,14 @@ public static void analyze(CreateDbStmt statement, ConnectContext context) {
         } catch (AnalysisException e) {
             ErrorReport.reportSemanticException(ErrorCode.ERR_BAD_CATALOG_ERROR, catalogName);
         }
+
+        Map<String, String> properties = statement.getProperties();
+        if (properties.containsKey(PropertyAnalyzer.PROPERTIES_STORAGE_VOLUME)) {
+            String volume = properties.get(PropertyAnalyzer.PROPERTIES_STORAGE_VOLUME);
+            if (RunMode.getCurrentRunMode() == RunMode.SHARED_NOTHING && !StorageVolumeMgr.LOCAL.equalsIgnoreCase(volume)) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_COMMON_ERROR,
+                        "Storage volume can only be local in shared nothing mode");
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateDbAnalyzer.java:52]:
```suggestion
                        "Storage volume can only be 'local' in shared nothing mode");
```

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateDbAnalyzer.java|4|5|80.00%|[51]|

**Review by wyb — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateDbAnalyzer.java|4|5|80.00%|[51]|

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateDbAnalyzer.java|4|5|80.00%|[51]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**